apps/www/components/Climatelab/Onboarding/ClimateProfile.js

```diff
@@ -0,0 +1,203 @@
+import { Component, Fragment } from 'react'
+import { Mutation } from '@apollo/client/react/components'
+import { gql } from '@apollo/client'
+import { css, merge } from 'glamor'
+
+import { TESTIMONIAL_IMAGE_SIZE } from '../../constants'
+import ErrorMessage from '../../ErrorMessage'
+import Portrait from '../../Profile/Portrait'
+import { ListedCheckbox } from '../../Profile/Settings'
+import { mutation } from '../../Profile/Edit'
+import UsernameField from '../../Profile/UsernameField'
+import Section from '../../Onboarding/Section'
+import withT from '../../../lib/withT'
+
+import {
+  Interaction,
+  Button,
+  FieldSet,
+  InlineSpinner,
+  mediaQueries,
+} from '@project-r/styleguide'
+
+const { P } = Interaction
+
+const PORTRAIT_SIZE_M = TESTIMONIAL_IMAGE_SIZE
+const PORTRAIT_SIZE_S = TESTIMONIAL_IMAGE_SIZE * 0.75
+
+const styles = {
+  p: css({
+    marginBottom: 20,
+  }),
+  field: css({
+    marginTop: 10,
+    marginBottom: 10,
+  }),
+  checkbox: css({
+    marginTop: 40,
+    marginBottom: 40,
+  }),
+  portrait: css({
+    width: PORTRAIT_SIZE_S,
+    height: PORTRAIT_SIZE_S,
+    [mediaQueries.mUp]: {
+      width: PORTRAIT_SIZE_M,
+      height: PORTRAIT_SIZE_M,
+    },
+  }),
+  actions: css({
+    marginBottom: 20,
+    display: 'flex',
+    flexWrap: 'wrap',
+    position: 'relative',
+    '& > div': {
+      flexGrow: 1,
+      margin: '5px 15px 0 0',
+      minWidth: '120px',
+      [mediaQueries.mUp]: {
+        flexGrow: 0,
+        margin: '5px 15px 0 0',
+        minWidth: '160px',
+      },
+    },
+  }),
+  save: css({
+    width: 160,
+    textAlign: 'center',
+  }),
+}
+
+export const fragments = {
+  user: gql`
+    fragment ProfileUser on User {
+      id
+      firstName
+      lastName
+      portrait
+      statement
+      username
+      isAdminUnlisted
+      isListed
+      isEligibleForProfile
+      hasPublicProfile
+    }
+  `,
+}
+
+class ClimateProfile extends Component {
```

Comment:
What's the reason for using a class component here?

Reply:
lazyness, i mostly copy pasted the code from Profile